### PR TITLE
forward plugin

### PR DESCRIPTION
**Pull request by skob**
some times it's needed to forward alert between alerta-servers

### CODE REVIEW

File: plugins/slack/alerta_slack.py [hunk 1]
```diff
@@ -61,6 +61,23 @@ def _slack_prepare_payload(self, alert, status=None, text=None):
...
+            color = "#FFFF00"  # yellow
+        elif alert.severity == 'warning':
+            color = "#1E90FF"  # blue
+        elif alert.severity == 'information':
+            color = "#808080"  # gray
+        elif alert.severity == 'debug':
+            color = "#808080"  # gray
+        elif alert.severity == 'trace':
+            color = "#808080"  # gray
+
```
Comment (satterly) [inline plugins/slack/alerta_slack.py]: I don't think you meant to include these changes.

File: plugins/forward/setup.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
+from setuptools import setup, find_packages
+
+version = '5.3.3'
```
Comment (satterly) [inline plugins/forward/setup.py]: Maybe reset this version number to `5.0.0` ?

File: plugins/forward/alerta_forward.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+LOG = logging.getLogger('alerta.plugins.forward')
+
+FORWARD_URL = app.config.get('FORWARD_URL')
+FORWARD_API = app.config.get('FORWARD_API')
+
+class ForwardAlert(PluginBase):
+
+    def pre_receive(self, alert):
+        return
```
Comment (satterly) [inline plugins/forward/alerta_forward.py]: The `pre-receive()` hook needs to return the alert like this ...
https://github.com/alerta/alerta/blob/master/alerta/plugins/__init__.py#L20

File: plugins/forward/alerta_forward.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import logging
+
+from alerta.plugins import PluginBase
+from alertaclient.api import Client
+
+LOG = logging.getLogger('alerta.plugins.forward')
+
+FORWARD_URL = app.config.get('FORWARD_URL')
+FORWARD_API = app.config.get('FORWARD_API')
```
Comment (satterly) [inline plugins/forward/alerta_forward.py]: Can we call the variable `FORWARD_API_KEY` to be consistent with other plugins? And we should allow them to be set from the shell environment as other plugins do -- especially as you say this is possible in the README.

File: plugins/forward/alerta_forward.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+class ForwardAlert(PluginBase):
+
+    def pre_receive(self, alert):
+        return
+
+    def post_receive(self, alert):
+        if not FORWARD_URL or not FORWARD_API:
+            return
+        client = Client(key=FORWARD_API, endpoint=FORWARD_URL)
```
Comment (satterly) [inline plugins/forward/alerta_forward.py]: Minor point, but can you swap the order of `endpoint` and `key` to be consistent with the method signature?

File: plugins/forward/alerta_forward.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        client.send_event(
+            event=alert.get('event'),
+            resource=alert.get('resource'),
+            alert
```
Comment (satterly) [inline plugins/forward/alerta_forward.py]: Are you sure this works? I would have thought you'd get an "multiple values for keyword argument 'event'" error.

**Comment by satterly** [comment]
What happens if two Alerta servers forward alerts to each other? Do you get the same alerts going backwards and forwards until the whole thing blows up? How do you prevent a forwarded alert from being forwarded back? And what about operator actions like `ack` alerts? Would it be expected that those actions should be forwarded as well?

**Comment by skob** [comment]
hmmm.
you right , i'll rework code for dedups and for your comments

In my opinion operator actions cannot be forwarded (and blackouts on start point should stop forwarding too)

**Pull request closed without merging**